Kernel/Arch/aarch64/PSCI.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2025, the SerenityOS developers
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/Arch/aarch64/PSCI.h>
+#include <Kernel/Firmware/DeviceTree/DeviceTree.h>
+
+namespace Kernel::PSCI {
+
+#define PSCI_FUNCTION_BASE 0x84000000
+
+#define PSCI_SYSTEM_OFF ((PSCI_FUNCTION_BASE) + 8)
+#define PSCI_SYSTEM_RESET ((PSCI_FUNCTION_BASE) + 9)
+
+static inline FlatPtr aarch64_hvc(FlatPtr arg0, FlatPtr arg1, FlatPtr arg2, FlatPtr arg3, FlatPtr arg4, FlatPtr arg5, FlatPtr arg6, FlatPtr arg7)
+{
+    register FlatPtr a0 asm("x0") = arg0;
+    register FlatPtr a1 asm("x1") = arg1;
+    register FlatPtr a2 asm("x2") = arg2;
+    register FlatPtr a3 asm("x3") = arg3;
+    register FlatPtr a4 asm("x4") = arg4;
+    register FlatPtr a5 asm("x5") = arg5;
+    register FlatPtr a6 asm("x6") = arg6;
+    register FlatPtr a7 asm("x7") = arg7;
+
+    asm volatile("hvc\t0\n"
+                 : "=r"(a0), "=r"(a1), "=r"(a2), "=r"(a3)
+                 : "0"(a0),
+                 "r"(a1),
+                 "r"(a2),
+                 "r"(a3),
+                 "r"(a4),
+                 "r"(a5),
+                 "r"(a6),
+                 "r"(a7));
+
+    return a0;
+}
+
+bool is_supported()
+{
+    // https://www.kernel.org/doc/Documentation/devicetree/bindings/arm/psci.txt
+    auto psci = DeviceTree::get().get_child("psci"sv);
+    if (!psci.has_value())
+        return false;
+
+    // NOTE: We reject "arm,psci" on purpose, since these old devices don't have standardized function IDs.
+    if (!psci->is_compatible_with("arm,psci-0.2"sv) && !psci->is_compatible_with("arm,psci-1.0"sv))
+        return false;
+
+    auto method = psci->get_property("method"sv);
+    VERIFY(method.has_value());
```

Comment:
Please don't `VERIFY` here. Returning `false` and maybe printing a debug message seems nicer than crashing on invalid devicetrees.